File: 3151_special_array_i.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& x) {
        for (int y = 0; y < x.size() - 1; y++)
            if (x[y] % 2 == x[y + 1] % 2) return false;
        return true;
    }
};

File: 3151_special_array_i.java
class Solution {
    public boolean isArraySpecial(int[] z) {
        int j = z.length;
        for (int i = 0; i < j - 1; i++) 
            if ((z[i] % 2) == (z[i + 1] % 2)) return false;
        return true;
    }
}
